code/drasil-gen/lib/Language/Drasil/Dump.hs

```diff
@@ -0,0 +1,35 @@
+-- FIXME: Why is this a `Language` top-level module name?
+module Language.Drasil.Dump where
+
+import qualified Database.Drasil as DB
+import SysInfo.Drasil (SystemInformation(_sysinfodb))
+
+import System.Directory
+import System.IO
+import Data.Aeson
+import Data.Aeson.Encode.Pretty
+import qualified Data.ByteString.Lazy.Char8 as LB
+import qualified Data.Map.Strict as SM
+
+import Utils.Drasil (invert)
+
+type Path = String
+type TargetFile = String
+
+dumpEverything :: SystemInformation -> Path -> IO ()
+dumpEverything si targetPath = do
+  createDirectoryIfMissing True targetPath
+  let chunkDump = DB.dumpChunkDB $ _sysinfodb si
+  let invertedChunkDump = invert chunkDump
+  let sharedUIDs = SM.filter (\x -> length x > 1) invertedChunkDump
```

Comment:
This is actually going to force `invertedChunkDump` to be fully computed, and then lots thrown away, because of `length` being strict.

```haskell
longerThan1 :: [a] -> Bool
longerThan2 (_ : _ : _) = True
longerThan2 _ = False
```
will only force the first 2 elements (if they exist) and not the rest.

Reply:
Ah, good catch! Thank you 79dbe4b960924b852503ef4a972cc29526fc6bcb